source/Agents/AgentLibary.hpp

```diff
@@ -15,250 +20,248 @@
+                        } else {
+                            for (int i = 0; i < steps; ++i) {
+                                positions.push_back(base_pos.Below());
+                            }
+                        }
+                        break;
+                    }
+
+                    // Move left
+                    case 'w': {
+                        if (multiply) {
+                            positions.push_back(base_pos.ToLeft(steps));
+                        } else {
+                            for (int i = 0; i < steps; ++i) {
+                                positions.push_back(base_pos.ToLeft());
+                            }
+                        }
+                        break;
+                    }
+
+                    // Move right
+                    case 'e': {
+                        if (multiply) {
+                            positions.push_back(base_pos.ToRight(steps));
+                        } else {
+                            for (int i = 0; i < steps; ++i) {
+                                positions.push_back(base_pos.ToRight());
+                            }
+                        }
+                        break;
+                    }
+
+                    // Stay
+                    case 'x': {
+                        // Using the `*` does nothing to scale the offset since it's scaling {0,
+                        // 0}
+                        steps = multiply ? 1 : steps;
+
+                        for (int i = 0; i < steps; ++i) {
+                            positions.push_back(base_pos);
+                        }
+                    }
+                }
+            } else {
+                std::ostringstream what;
+                what << "Incorrectly formatted argument: " << single_command;
+                throw std::invalid_argument(what.str());
             }
-          }
-          break;
-        }
 
-          // Stay
-        case 'x': {
-          // Using the `*` does nothing to scale the offset since it's scaling {0,
-          // 0}
-          steps = multiply ? 1 : steps;
-
-          for (int i = 0; i < steps; ++i) {
-            positions.push_back(base_pos);
-          }
+            iss >> std::skipws;
         }
-      }
-    } else {
-      std::ostringstream what;
-      what << "Incorrectly formatted argument: " << single_command;
-      throw std::invalid_argument(what.str());
+        return positions;
     }
 
-    iss >> std::skipws;
-  }
-  return positions;
-}
-
-template<typename T>
-concept Agent_Type = std::is_base_of_v<cse491::AgentBase, T>;
+    template<typename T>
+    concept Agent_Type = std::is_base_of_v<cse491::AgentBase, T>;
 
-template<typename T>
-T &DownCastAgent(cse491::Entity &entity) requires(Agent_Type<T>) {
-  assert(dynamic_cast<T *>(&entity)!=nullptr);
-  return static_cast<T &>(entity);
-}
+    template<typename T>
+    T &DownCastAgent(cse491::Entity &entity) requires(Agent_Type<T>) {
```

Comment:
A brief description (one line) would be good for completeness, or anything that we have defined ourselves. For this one, it's a helper that simplifies the boilerplate of having to always downcast entities that have been added to the world